src/Param.cc

```diff
@@ -870,6 +906,47 @@ bool PoseStringFromValue(const PrintConfig &_config,
     _valueStr = ss.str();
     return true;
   }
+  else if (rotationFormat == "euler_rpy" && inDegrees && snapDegreesToInterval)
+  {
+    // Helper function that returns a snapped value if it is within the
+    // tolerance of multiples of interval, otherwise the orginal value is
+    // returned.
+    auto snapToInterval =
+        [](double _val, unsigned int _interval, double _tolerance)
+    {
+      const double sign = (_val < 0) ? -1 : 1;
+      unsigned int quotient = static_cast<unsigned int>(abs(_val)) / _interval;
+
+      double lowerRemainder =
+          abs(_val) - static_cast<double>(quotient * _interval);
+      if (lowerRemainder< _tolerance)
+      {
+        return static_cast<double>(_interval * quotient) * sign;
+      }
+
+      double higherRemainder =
+          static_cast<double>(++quotient * _interval) - abs(_val);
+      if (higherRemainder < _tolerance)
+      {
+        return static_cast<double>(_interval * quotient) * sign;
+      }
+
+      return _val;
```

Comment:
At the risk of sounding pedantic, I want to suggest a different implementation. The issue I see here is if that if the threshold is large enough, this logic would pick the wrong value. For example, if the snap value is 5 and threshold is 4 and the angle is 48.5, the logic picks 45, but I would expect it to pick 50, since it's the closest to 48.5. The implementation I'm suggesting first picks the quotient that has the smallest remainder by rounding the quotient.

```suggestion
      double closestQuotient = std::round(_val / _interval);
      double distance = std::abs(_val - closestQuotient * _interval);
      if (distance < _tolerance)
      {
        return _interval * closestQuotient;
      }
      return _val;
```

Reply:
Thanks for spotting that! I used your suggested implementation and added a test, 
https://github.com/ignitionrobotics/sdformat/pull/689/commits/236e8583631844994418302e4a2498b1b3a10934